Sources/Bot/Services/MergeService.swift

```diff
@@ -147,24 +321,22 @@ extension MergeService {
         })
     }
 
-    fileprivate static func whenStarting(github: GitHubAPIProtocol, scheduler: Scheduler) -> Feedback<State, Event> {
+    fileprivate static func whenStarting(initialPullRequests: [PullRequest], scheduler: DateScheduler) -> Feedback<State, Event> {
         return Feedback(predicate: { $0.status == .starting }) { state -> SignalProducer<Event, NoError> in
-
-            return github.fetchPullRequests()
-                .flatMapError { _ in .value([]) }
-                .map { pullRequests in
-                    pullRequests.filter { $0.isLabelled(with: state.integrationLabel) }
-                }
-                .map(Event.pullRequestsLoaded)
-                .start(on: scheduler)
+            return SignalProducer
+                .value(Event.pullRequestsLoaded(initialPullRequests))
+                .observe(on: scheduler)
         }
     }
 
     fileprivate static func whenReady(github: GitHubAPIProtocol, scheduler: Scheduler) -> Feedback<State, Event> {
         return Feedback(predicate: { $0.status == .ready }) { state -> SignalProducer<Event, NoError> in
 
-            guard let next = state.pullRequests.first
-                else { return .value(.noMorePullRequests) }
+            guard let next = state.pullRequests.first else {
+                return SignalProducer
+                    .value(.noMorePullRequests)
+                    .observe(on: scheduler)
+            }
```

Comment:
This missing `.observe(on: scheduler)` is note related to my changes directly but was the one causing the race condition on my tests (but only on Linux, so that was not easy to debug). Took a while to find it 😅